src/test/java/org/jenkinsci/plugins/workflow/WorkflowJobNonRestartingTest.java

```diff
@@ -33,31 +33,31 @@
 import hudson.util.OneShotEvent;
 import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted;
 import org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval;
-import org.jenkinsci.plugins.workflow.cps.AbstractCpsFlowTest;
 import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
 import org.jenkinsci.plugins.workflow.job.WorkflowJob;
 import org.jenkinsci.plugins.workflow.job.WorkflowRun;
 import static org.junit.Assert.*;
-import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.jvnet.hudson.test.Issue;
 
 import org.junit.ClassRule;
 import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.JenkinsRule;
 
 /**
  * Test of {@link WorkflowJob} that doesn't involve Jenkins restarts.
  *
  * @author Kohsuke Kawaguchi
  */
-public class WorkflowJobNonRestartingTest extends AbstractCpsFlowTest {
+public class WorkflowJobNonRestartingTest {
 
     @ClassRule public static BuildWatcher buildWatcher = new BuildWatcher();
+    @ClassRule public static JenkinsRule jenkins = new JenkinsRule();
 
-    WorkflowJob p;
+    private static WorkflowJob p;
 
-    @Before public void setUp() throws Exception {
-        super.setUp();
+    @BeforeClass public static void setUp() throws Exception {
         p = jenkins.jenkins.createProject(WorkflowJob.class, "demo");
```

Comment:
Note that this will reuse the same project each time. Probably fine, but it might be better to delete this method and add

```java
WorkflowJob p = jenkins.createProject(WorkflowJob.class);
```

to the start of every test method. Or just revert the change to the field and to the signature of this method and merely delete the `name` argument.

Reply:
Good point, I switched to creating a project in every test method in ea3cd8a.